### PR TITLE
 Wrap doc update function with try-catch

### LINKED ISSUE #77: TypeError: Cannot read property 'getValue' of undefined
Sentry Issue: [CODEPAIR-6](https://sentry.io/organizations/yorkie-team/issues/2283904124/?referrer=github_integration)

```
TypeError: Cannot read property 'getValue' of undefined
  at get (../yorkie/src/document/json/rga_tree_list.ts:207:56)
  at get (../yorkie/src/document/json/array.ts:57:26)
  at getElementByID (../yorkie/src/document/proxy/array_proxy.ts:54:44)
  at updater (components/Editor/Board/Canvas/worker.ts:69:41)
  at update (../yorkie/src/document/document.ts:126:7)
...
(3 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by ppeeou**
#### What this PR does / why we need it?

Wrap yorkie doc update function with try-catch
 
why?

When using  `deleteByID` `getElementByID`, an error occurs if the node has been deleted by GC.

#### What are the relevant tickets?

Fixes #77 

### Checklist
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for the work.
I am concerned that the scope of `tryIt` is too broad to ignore other errors. And I think it's better to wrap only those methods where we want to intentionally ignore errors.

**Comment by hackerwins** [comment]
@ppeeou 
Please re-sign me when the correction is complete.

File: .eslintrc.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
     "react/jsx-props-no-spreading": 0,
     "react/jsx-one-expression-per-line": 0,
     "class-methods-use-this": 0,
-    "import/prefer-default-export": 0
+    "import/prefer-default-export": 0,
+    "no-useless-return": 0,
+    "consistent-return": 0
```
Comment (ppeeou) [inline .eslintrc.json:15]: The reason for adding it is as follows.

We're already getting help from typescript, so I think the above configuration is superfluous.🤔

### no-useless-return

image

If the return statement is deleted, the following error occurs
image


### consistent-return
image

File: src/components/Editor/DrawingBoard/Canvas/Worker/Worker.ts [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
+      return;
+    }
+  }
+
+  deleteByID(root: Root, createID: TimeTicket): Shape | undefined {
+    try {
+      root.shapes.deleteByID(createID);
+    } catch {
+      return;
+    }
```
Comment (hackerwins) [inline src/components/Editor/DrawingBoard/Canvas/Worker/Worker.ts:35]:
```suggestion
    try {
      return root.shapes.deleteByID(createID);
    } catch {
      return undefined;
    }
```

File: src/components/Editor/DrawingBoard/Canvas/Worker/Worker.ts [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
+
+  abstract flushTask(): void;
+
+  getElementByID(root: Root, createID: TimeTicket): Shape | undefined {
+    try {
+      const lastShape = root.shapes.getElementByID(createID);
+      return lastShape;
+    } catch {
+      return;
+    }
```
Comment (hackerwins) [inline src/components/Editor/DrawingBoard/Canvas/Worker/Worker.ts:27]:
```suggestion
    try {
      return root.shapes.getElementByID(createID);
    } catch {
      return undefined;
    }
```

File: .eslintrc.json (hunk 1 above)
Comment (hackerwins) [inline .eslintrc.json:15]:
```suggestion
    "import/prefer-default-export": 0
```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thank you for the detailed description. Based on your explanation, I've confirmed that when I am modifying the code like below it doesn't cause any problems even if I don't remove the rule.

**Comment by ppeeou** [comment]
> Thank you for the detailed description. Based on your explanation, I've confirmed that when I am modifying the code like below it doesn't cause any problems even if I don't remove the rules.

@hackerwins  Thank you for your comments!
In javascript, if the return value does not exist from a function call, it returns `undefined`. Since typescript is inferring this part, I don't think we need to explicitly use 'undefined', so I added the above condition..!

File: src/components/Editor/DrawingBoard/Canvas/Worker/Worker.ts (hunk 2 above)
Comment (hackerwins) [reply src/components/Editor/DrawingBoard/Canvas/Worker/Worker.ts:35]: Even though `root.shapes.deleteByID` returns the deleted element, is there any reason why this method didn't return that element?

**Review by hackerwins — CHANGES_REQUESTED** [review]
@ppeeou 

I think the rules we are trying to eliminate here are meaningful and help reduce mistakes. Wouldn't it be better to remove these rules because we don't want to express `return undefined`?

**Comment by ppeeou** [comment]
> Even though root.shapes.deleteByID returns the deleted element, is there any reason why this method didn't return that element?

There was no place to use the return value but It would be better to return it as you said :) I fixed it

**Comment by hackerwins** [comment]
@ppeeou I'm still hesitating whether removing the lint rules would be a good thing for us. 
Can you help me to make more sense?

**Comment by ppeeou** [comment]
> @ppeeou I'm still hesitating whether removing the lint rules would be a good thing for us.
> Can you help me to make more sense?

I think typescript is inferring the function return value (if undefined ) / (if there is a value). So I tried to remove the explicit use of undefined.

but If someone on the team wants to code more strictly and explicitly,
I'll process it without removing the lint.


method 1,2 don't have to delete the current lint rule
if use it like 3, need to remove the lint rule... my personal preference to use it like 3

### 1 
```js
deleteByID(root: Root, createID: TimeTicket): Shape | undefined {
  let shape;
  try {
    shape = root.shapes.deleteByID(createID);
    return shape;
  } catch {
    return shape;
  }
}
```

### 2
 ```js
deleteByID(root: Root, createID: TimeTicket): Shape | undefined {
    try {
      return root.shapes.deleteByID(createID);
    } catch {
      return undefined;
    }
  }
```

### 3
```js
deleteByID(root: Root, createID: TimeTicket): Shape | undefined { 
  try {
    return root.shapes.deleteByID(createID);
  } catch {
    return;
  }
}
```

**Comment by hackerwins** [comment]
@ppeeou I think it's better to be more strict with the help of tools(I like the 2nd one).
What are the opinions of @mojosoeun , @dc7303 ?

**Comment by dc7303** [comment]
@ppeeou @hackerwins  I agree with both of you and I think it's a reasonable idea. You've asked me for opinions, so I'll share my thoughts.

I personally think disabling lint rules should be avoided. And I think so in this case.

Returning a value from a function means using the returned value in the next logic. To be exaggerated, `return undefined;` is to specify the value to be used in the upper call stack for this function, and `return;` can be seen as the meaning of break. 

Maybe trivial, but sometimes I like code written strictly like this. Because I think it can reduce even the smallest mistakes.

**Comment by ppeeou** [comment]
@hackerwins @dc7303 Thanks for your comments,👍 I also agree that it should be written strictly.
I'm going to proceed in the direction of not removing lint!

**Comment by ppeeou** [comment]
@hackerwins @dc7303 Thanks for your comments,👍 I also agree that it should be written strictly.
I'm going to proceed in the direction of not removing lint!

**Review by hackerwins — APPROVED** [review]
LGTM 👍

**Pull request merged**